bindinfo/cache.go

```diff
@@ -191,3 +193,84 @@ func (b cache) appendNode(newBindRecord *bindRecord, sparser *parser.Parser) err
 	b[hash] = append(b[hash], newNode)
 	return nil
 }
+
+// AddGlobalBind implements GlobalBindAccessor.AddGlobalBind interface.
+func (h *Handle) AddGlobalBind(originSQL, bindSQL, defaultDB, charset, collation string) error {
+	h.lock.Lock()
+	defer h.lock.Unlock()
+
+	ctx := context.TODO()
+	exec, _ := h.ctx.(sqlexec.SQLExecutor)
+	_, err := exec.Execute(ctx, "BEGIN")
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err == nil {
+			_, err = exec.Execute(ctx, "COMMIT")
+		} else {
+			_, rbErr := exec.Execute(ctx, "ROLLBACK")
+			terror.Log(rbErr)
+		}
+	}()
+
+	sql := fmt.Sprintf("SELECT _tidb_rowid,status FROM mysql.bind_info WHERE original_sql='%s' AND default_db='%s'",
+		originSQL, defaultDB)
+	rs, err := exec.Execute(ctx, sql)
+	if err != nil {
+		return err
+	}
+	if len(rs) >= 1 {
+		chkBatch := rs[0].NewRecordBatch()
+		for {
+			err = rs[0].Next(context.TODO(), chkBatch)
+			if err != nil {
+				return err
+			}
+			if chkBatch.NumRows() == 0 {
+				break
+			}
+			it := chunk.NewIterator4Chunk(chkBatch.Chunk)
+			for row := it.Begin(); row != it.End(); row = it.Next() {
+				rowID := row.GetInt64(0)
+				status := row.GetString(1)
+				if status == using {
+					return errors.New("origin sql already has binding sql")
+				}
+				sql = fmt.Sprintf("DELETE FROM mysql.bind_info WHERE _tidb_rowid=%d", rowID)
+				_, err = exec.Execute(ctx, sql)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	txn, err := h.ctx.Txn(true)
+	if err != nil {
+		return err
+	}
+	ts := getTimeStringWithoutZone(oracle.GetTimeFromTS(txn.StartTS()))
+	bindSQL = getEscapeCharacter(bindSQL)
+	sql = fmt.Sprintf(`INSERT INTO mysql.bind_info(original_sql,bind_sql,default_db,status,create_time,update_time,charset,collation) VALUES ('%s', '%s', '%s', '%s', '%s', '%s','%s', '%s')`,
+		originSQL, bindSQL, defaultDB, using, ts, ts, charset, collation)
+	_, err = exec.Execute(ctx, sql)
+	return err
+}
+
+func getTimeStringWithoutZone(ts time.Time) string {
+	tsString := ts.String()
```

Comment:
Can `ts.Format` help? If so, we do not need `getTimeStringWithoutZone`

Reply:
`ts.Format` can't process this case, the format result is incorrect